mkosi/distributions/gentoo.py

```diff
@@ -104,78 +101,49 @@ def install(cls, state: MkosiState) -> None:
-        pkgmngr_config = state.pkgmngr / "etc/portage"
-        root_portage_cfg = state.root / "etc/portage"
-        root_portage_cfg.mkdir(parents=True, exist_ok=True)
-
-        if not stage3_tar_path.exists():
-            if stage3_cache.exists():
-                log_step('New stage3 is available , removing cache')
-                unlink_try_hard(state.cache_dir.joinpath(stage3_tar).parent)
-                unlink_try_hard(stage3_cache)
-            if vanilla_config.exists():
-                unlink_try_hard(vanilla_config)
-            with complete_step(f"Fetching {stage3_url_path}"):
-                stage3_tar_path.parent.mkdir(parents=True, exist_ok=True)
-                urllib.request.urlretrieve(stage3_url_path, stage3_tar_path)
-        stage3_cache.mkdir(parents=True, exist_ok=True)
-
-        if next(stage3_cache.iterdir(), None) is None:
-            with complete_step(f"Extracting {stage3_tar.name} to {stage3_cache}"):
+        stage3.mkdir(exist_ok=True)
+
+        if not any(stage3.iterdir()):
+            with complete_step(f"Extracting {stage3_tar.name} to {stage3}"):
                 run([
                     "tar",
                     "--numeric-owner",
-                    "-C", stage3_cache,
+                    "-C", stage3,
                     "--extract",
-                    "--file", stage3_tar_path,
-                    "--exclude", "./dev",
-                    "--exclude", "./proc",
+                    "--file", stage3_tar,
+                    "--exclude", "./dev/*",
+                    "--exclude", "./proc/*",
+                    "--exclude", "./sys/*",
                 ])
-            copy_path(config, vanilla_config)
-
-        # why can't we use --config-root or PORTAGE_CONFIGROOT via
-        # invoke_emerge()?
-        #
-        # from emerge(1)
-        # PORTAGE_CONFIGROOT is now superseded by the SYSROOT variable and
-        # can only be given if its value matches SYSROOT or if ROOT=/.
-        # Defaults to / .
-        unlink_try_hard(config)
-        if pkgmngr_config.exists():
-            copy_path(pkgmngr_config, config)
-        else:
-            copy_path(vanilla_config, config)
-        copy_path(config, root_portage_cfg)
 
         for d in ("binpkgs", "distfiles", "repos"):
-            state.cache_dir.joinpath(d).mkdir(exist_ok=True)
-
-        bwrap_params: list[PathString] = [
-            "--bind", state.cache_dir / "repos", "/var/db/repos"
-        ]
-        run_workspace_command(stage3_cache, ["/usr/bin/emerge-webrsync"],
-                              bwrap_params=bwrap_params, network=True)
-
-        with complete_step("Layingout basic filesystem"):
-            invoke_emerge(state, options=["--emptytree"],
-                          packages=["sys-apps/baselayout"],
-                          env={'USE': 'build'})
+            (state.cache_dir / d).mkdir(exist_ok=True)
+
+        copy_path(state.pkgmngr, stage3, preserve_owner=False)
```

Comment:
image needs these configs, most importantly `make.profile`. we probably want `copy_path(config, root_portage_cfg)` back.

Reply:
If I look inside /etc/portage in the stage3 directory in mkosi.cache, make.profile is a dangling symlink. The only actual file in /etc/portage is make.conf with the following contents:

```
# These settings were set by the catalyst build script that automatically
# built this stage.
# Please consult /usr/share/portage/config/make.conf.example for a more
# detailed example.
COMMON_FLAGS="-O2 -pipe"
CFLAGS="${COMMON_FLAGS}"
CXXFLAGS="${COMMON_FLAGS}"
FCFLAGS="${COMMON_FLAGS}"
FFLAGS="${COMMON_FLAGS}"

# NOTE: This stage was built with the bindist Use flag enabled

# This sets the language of build output to English.
# Please keep this setting intact when reporting bugs.
LC_MESSAGES=C.utf8
```

I don't think users get much value from us copying this into the image. Anyone that wants an updateable gentoo image will install portage in the image and provide custom configuration files in mkosi.skeleton/ which will be included in the image so there should be no need for us to copy these to the image.

Reply:
> then they will have a drift no? they have to keep a copy of the config in mkosi.skeleton and also in pkgmng.
also i suppose you mean mkosi.extra because then we're mixing stuff in skeleton and pkgmng, it will be a hell to maintain for users.

By default, unless --package-manager-tree is specified explicitly, mkosi.skeleton/ is used. So there won't be any drift as the skeleton tree and the package manager tree will be the exact same tree, so the same configs that are used to build the image will be copied into the image as well with no drift.